### PR TITLE
First Training script for SSL added

### PR DESCRIPTION
**Pull request by finalelement**
Signed-off-by: vnath <vnath@nvidia.com>

### Description
This PR is the first step to integrating our 2-step tutorial for Self-supervised learning. The added script works on the TCIA Covid-19 data.

### Status
**Work in progress**

### Checks

- [ ] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

File: self_supervised_pretraining/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+### 1.Data
+Pretraining Dataset: The TCIA Covid-19 dataset was used for generating the pretrained weights.
+The dataset contains a total of 771 3D CT Volumes. The volumes were split into training and validation sets
+of 600 and 171 3D volumes correspondingly. PUT LINKS & CITATIONS HERE
+
+Finetuning Dataset: The dataset from Beyond the Cranial Vault Challenge 2015 hosted at MICCAI was used as a
+fully supervised Finetuning task on the pre-trained weights. The dataset contains of 30 3D Volumes with annotated labels
+for 13 different organs. PUT LINKS & CITATIONS HERE
+
+### 2. Network Architectures
```
Comment (Nic-Ma) [inline self_supervised_pretraining/README.md:19]: I think the items 2 - 5 are still not updated?

Thanks.

File: self_supervised_pretraining/ssl_script_train.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    RandSpatialCropd,
+    RandSpatialCropSamplesd,
+    RandCoarseDropoutd,
+    RandCoarseShuffled
+)
+
+def main():
+
+    # TODO Defining file paths
+    #json_Path = os.path.normpath('/scratch/data_2021/tcia_covid19/dataset_split_debug.json')
```
Comment (Nic-Ma) [inline self_supervised_pretraining/ssl_script_train.py:34]: please remove some debug issue.

Thanks.

File: self_supervised_pretraining/ssl_script_train.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            2.0, 2.0, 2.0), mode=("bilinear")),
+        ScaleIntensityRanged(
+            keys=["image"], a_min=-57, a_max=164,
+            b_min=0.0, b_max=1.0, clip=True,
+        ),
+        CropForegroundd(keys=["image"], source_key="image"),
+        SpatialPadd(keys=["image"], spatial_size=(96, 96, 96)),
+        RandSpatialCropSamplesd(keys=["image"], roi_size=(96, 96, 96), random_size=False, num_samples=2),
+        #RandSpatialCropd(keys=["image"], roi_size=(96, 96, 96), random_size=False),
+        #TODO Ask Wenqi, if I create another copy of the image using CopyItemsd and then pass both 'image_1', 'image_2'
```
Comment (Nic-Ma) [inline self_supervised_pretraining/ssl_script_train.py:78]: Please remove these commens.

Thanks.

File: self_supervised_pretraining/ssl_script_train.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    print('#######################')
+    print('Total Number of Validation Data Samples: {}'.format(len(val_Data)))
+    print(val_Data)
+    print('#######################')
+
+    ###############End Of Append Paths#################
+    # TODO Set Determinism
+    set_determinism(seed=453)
+
+    # TODO Define Training Transforms
```
Comment (Nic-Ma) [inline self_supervised_pretraining/ssl_script_train.py:63]: Same as above comment.

Thanks.

File: self_supervised_pretraining/ssl_script_train.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+    )
+
+    check_ds = Dataset(data=train_Data, transform=train_Transforms)
+    check_loader = DataLoader(check_ds, batch_size=1)
+    check_data = first(check_loader)
+    orig_image = (check_data["gt_image"][0][0])
+    image = (check_data["image"][0][0])
+    print(f"image shape: {image.shape}")
+
+    '''
```
Comment (Nic-Ma) [inline self_supervised_pretraining/ssl_script_train.py:99]: Please remove these test code.

Thanks.

**Comment by finalelement** [comment]
@Nic-Ma Thank you for the wonderful comments Nic, Ill take care of them (all minor stuff). The stoppers for me regarding this snippet of code are:
- How to get rid of paths Line 35, 36, 37 (where can I put the data and what paths I can use to get to the data etc)
- What paths should I use to save the model check point and final plotting of results?

File: self_supervised_pretraining/README.md (hunk 1 above)
Comment (finalelement) [reply self_supervised_pretraining/README.md:19]: Yes, the objective for now is to outline all broad things that need to be covered. It will be addressed.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Could you please help give some advice to store data?

Thanks in advance.

File: self_supervised_pretraining/ssl_script_train.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    check_ds = Dataset(data=train_Data, transform=train_Transforms)
+    check_loader = DataLoader(check_ds, batch_size=1)
+    check_data = first(check_loader)
+    orig_image = (check_data["gt_image"][0][0])
+    image = (check_data["image"][0][0])
+    print(f"image shape: {image.shape}")
+
+    # Define Network ViT backbone & Loss & Optimizer
+    device = torch.device("cuda:0")
+    model = ViT(
```
Comment (wyli) [inline self_supervised_pretraining/ssl_script_train.py:95]: is the recent `ViTAutoEnc` not used in this SSL tutorial?

**Comment by finalelement** [comment]
@wyli @Nic-Ma please check out the 'ssl_script_train.py' and provide me feedback, I updated it with all MONAI components that we had been working on. 

Also tested it locally, it was running as expected on a small dataset. Please let me know if that script can be improved.

File: self_supervised_pretraining/ssl_script_train.py (hunk 5 above)
Comment (finalelement) [reply self_supervised_pretraining/ssl_script_train.py]: This is removed.

File: self_supervised_pretraining/README.md (hunk 1 above)
Comment (finalelement) [reply self_supervised_pretraining/README.md:41]: These items have been updated.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**